### PR TITLE
test-servers: enable audit logs

### PR DESCRIPTION
**Pull request by sttts**
Enables us to 
- trace the whole life-cycle of every object. I.e. which state at which time, changed by which actor. Everything.
- identify hotlooping or just which component produces most load.

### CODE REVIEW

**Comment by s-urbaniak** [comment]
nice!

**Comment by sttts** [comment]
Fixed the rebase error 🤞

**Comment by sttts** [comment]
Reduced log by a factor of 3. Rougly we should have 170mb of audit logs, but compressable to around 10%.

File: cmd/test-server/kcp/audit-policy.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+- level: Metadata
+  verbs: ["get", "delete"]
+  omitStages:
+  - ResponseStarted
+
+- level: RequestResponse
+  verbs: ["create", "update", "patch"]
+  omitStages:
+  - ResponseStarted
```
Comment (s-urbaniak) [inline cmd/test-server/kcp/audit-policy.yaml:30]: can we shortly describe why these policy settings are being used, why they are useful in tests?

File: pkg/server/controllers.go [hunk 2]
```diff
@@ -124,7 +124,7 @@ func (s *Server) installKubeNamespaceController(ctx context.Context, config *res
 		metadata,
 		discoverResourcesFn,
 		s.kubeSharedInformerFactory.Core().V1().Namespaces(),
-		time.Duration(30)*time.Second,
+		time.Duration(5)*time.Minute,
```
Comment (s-urbaniak) [inline pkg/server/controllers.go:127]: why does this need to be bumped 10 times?

Comment (s-urbaniak) [reply pkg/server/controllers.go:127]: maybe a small comment?

Comment (s-urbaniak) [reply pkg/server/controllers.go:127]: nm, the commit message does say it all

File: cmd/test-server/kcp/audit-policy.yaml (hunk 1 above)
Comment (s-urbaniak) [reply cmd/test-server/kcp/audit-policy.yaml:30]: not a blocker

Comment (sttts) [reply cmd/test-server/kcp/audit-policy.yaml:30]: you can trace the whole life-cycle of every object. I.e. which state at which time, changed by which actor. Everything.

Comment (sttts) [reply cmd/test-server/kcp/audit-policy.yaml:30]: added to description

**Comment by sttts** [comment]
/ignore ci/prow/e2e-multiple-runs

Auditing is not enabled in e2e-multiple-runs. We have too many flakes in there.

**Comment by sttts** [comment]
/override ci/prow/e2e-multiple-runs

ci/prow/e2e-multiple-runs does not use auditing, only shared and sharded tests do.

**Pull request merged**